src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -100,8 +100,10 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
     if (messageStr.equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)
         || messageStr.equals(HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE)) {
       if (nameDir.exists()) {
+        // TODO (elingg) test failure scenario to make sure NN restarts with the
+        // appropriate sequence of events
         log.info(String
-            .format("NameNode data directory %s already exists, not formatting just starting",
+            .format("NameNode data directory %s already exists, not formatting",
```

Comment:
Yep, this needs to be handled with the processes starting and being reinitialized. The TODO is there, and this will be covered in an upcoming PR to ensure NN restart works correctly on the same node.